docs/cookbook/deploying-to-heroku.md

```diff
@@ -0,0 +1,222 @@
+
+Next, create an account at [heroku.com](http://heroku.com).
+
+Then create a Heroku app, choosing any app name and runtime location (US, Europe, etc.) you wish. While it is good to try and name your app the same as your project, this may not be possible if your app has a generic name like 'heroku-tut'. Remember the name of the app because we'll need it in future steps.
+
+Following the instructions on the Heroku site, install the [Heroku CLI](https://devcenter.heroku.com/articles/heroku-command-line) if you haven't already.
+
+*** From within your project directory log-in to heroku
+
+```bash
+$ heroku login
+```
+
+To enable deployment, add Heroku as a "git remote" to which your code can be pushed, substituting your Heroku app name created above:
+
+```bash
+$ heroku git:remote -a <yourAppName>
+```
+
+You can check to make sure this worked by using the command:
+
+```bash
+$ git remote -v
+```
+You should see output like, but with your app name at the end:
+
+```bash
+heroku  https://githeroku.com/apos-heroku-tut.git (fetch)
+heroku  https://githeroku.com/apos-heroku-tut.git (push)
+```
+
+## Add a MongoDB Atlas cluster
+With our project set-up to deploy to Heroku, we need a database.
+
+Heroku runs our node app, but it doesn't run MongoDB for us. So let's go to [MongoDB Atlas](https://www.mongodb.com/cloud/atlas) and sign up.
+
+Once you log-in to your account, create a project. I named my project 'apostrophe', but you can use any name you wish. Next, within that project, click on 'Build a Database'. Select the free'Shared' tier and pick 'AWS' as your cloud provider and the same region you chose for Heroku.
+For "Cluster Tier" choose "M0 Sandbox".
+
+We recommend you give your cluster the same name as your project, in the case of this example, 'heroku-tut'.
+
+You will need to set up an administrative MongoDB user for your cluster. These will be part of your MongoDB database connection credentials. **Be sure to set a secure, separate username and password,** do not use your Atlas login credentials. One thing to note: the characters `: / ? ! # [ ] @` in either your username or password will need to be [URL-encoded](https://www.eso.org/~ndelmott/url_encode.html) in the database connection string.
+
+### IP address allowing
+
+MongoDB Atlas requires us to allowlist the IP addresses that should be allowed to talk to the database. **Yes, it is secured by an encrypted password,** but this still helps to reduce potential DOS attacks.
+
+This is a problem with Heroku because it may connect from many IP addresses.
+
+If you are buying a larger Atlas plan you may be able to use the "VPC Peering" option, the details of which are beyond the scope of this recipe. Otherwise, use this IP address range:
+
+`0.0.0.0/0`
+
+This will allow you to connect for the purposes of this tutorial, but should be modified for a production site. One solution is to use the [fixie-socks](https://elements.heroku.com/addons/fixie-socks) add-on from Heroku.
```

Comment:
Honestly this is what we do for production too. I think you don't have to word it quite as strongly as "should be modified." Using fixie-socks could impact performance... there are pros and cons.

Reply:
Changed, but I could be persuaded to eliminate it.